### PR TITLE
Client Export

### PR DESCRIPTION
**Pull request by yakloinsteak**
Haven't been able to test against the production server. Still blocked.

### CODE REVIEW

**Review by eanders — APPROVED** [review]
I think this is good now.

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/client_export.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+        client_values = client_columns.map do |col|
+          client.send(col)
+        end
+
+        mci =
+          client.external_ids.to_a.find do |x|
+            x.namespace == HmisExternalApis::AcHmis::Mci::SYSTEM_ID
+          end
+        external_id_values = [mci&.value]
```
Comment (gigxz) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/client_export.rb]:
```suggestion
        # if the client has multiple MCI IDs, it doesn't matter which one we send
        external_id_values = [client.ac_hmis_mci_ids&.first]
```

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/client_export.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    def csv_config
+      {
+        write_converters: ->(value, _) {
+          if value.instance_of?(Date)
+            value.strftime('%Y-%m-%d')
+          elsif value.respond_to?(:strftime)
+            value.strftime('%Y-%m-%d %H:%M:%S')
+          else
+            # FIXME: Do the integer fields that represent boolean values default to zero? Or is nil/null correct?
```
Comment (gigxz) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/client_export.rb]: nil/null is correct

File: drivers/hmis_external_apis/lib/tasks/export.rake [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace :export do
+  # ./bin/rake driver:hmis_external_apis:export:ac_clients
+  desc 'Import AC client data'
```
Comment (gigxz) [inline drivers/hmis_external_apis/lib/tasks/export.rake]:
```suggestion
  desc 'Export AC client data'
```
maybe?

**Review by gigxz — APPROVED** [review]
Looks good! some minor comments inline

**Comment by yakloinsteak** [comment]
I'll try to wrap this up today.

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/client_export.rb (hunk 1 above)
Comment (yakloinsteak) [reply drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/client_export.rb]: Good idea. I've pushed something that captures that (had to make a few tiny changes)

**Comment by yakloinsteak** [comment]
I think I've addressed everything now. Maybe okay to merge after CI passes.

**Comment by gigxz** [comment]
Looks good @yakloinsteak but did you mean to push those ssh key files? Just checking..

**Comment by yakloinsteak** [comment]
Thanks for asking. Yes, they're only used in development and test, and they keep us from having to constantly deal with host keys changing each time we start up the sftp server.

**Pull request merged**